### DISCUSSION TITLE
The 'EnableCSI' feature flag was specified, but CSI API group [snapshot.storage.k8s.io/v1beta1] was not found

### DISCUSSION THREAD

**Discussion by avekrivoy** [vmware-tanzu/velero#4954] [Community support Q&A] [answered]
Hello! I'm using Rook as a storage provider and trying to enable CSI snapshots feature in Velero. Adding EnableCSI feature causes  Velero to fail during startup: 
```
time="2022-05-31T14:04:34Z" level=fatal msg="The 'EnableCSI' feature flag was specified, but CSI API group [snapshot.storage.k8s.io/v1beta1] was not found." logSource="pkg/cmd/server/server.go:551"
```
I followed instructions for the external snapshotter (https://github.com/kubernetes-csi/external-snapshotter)
```
Install Snapshot CRDs:

kubectl kustomize client/config/crd | kubectl create -f -
https://github.com/kubernetes-csi/external-snapshotter/tree/master/client/config/crd
Do this once per cluster
Install Common Snapshot Controller:

Update the namespace to an appropriate value for your environment (e.g. kube-system)
kubectl -n kube-system kustomize deploy/kubernetes/snapshot-controller | kubectl create -f -
Do this once per cluster
``` 
CSI driver was installed during Rook deployment as well:
```
kubectl get csidriver
NAME                            ATTACHREQUIRED   PODINFOONMOUNT   STORAGECAPACITY   TOKENREQUESTS   REQUIRESREPUBLISH   MODES        AGE
rook-ceph.cephfs.csi.ceph.com   true             false            false             <unset>         false               Persistent   124d
rook-ceph.rbd.csi.ceph.com      true             false            false             <unset>         false               Persistent   124d
```
What am I doing wrong? Is Velero's CSI snapshot feature compartible with recent Kubernetes versions (1.23)? 

```
kubectl api-resources | grep snapshot.storage.k8s.io
volumesnapshotclasses             vsclass,vsclasses   snapshot.storage.k8s.io/v1             false        VolumeSnapshotClass
volumesnapshotcontents            vsc,vscs            snapshot.storage.k8s.io/v1             false        VolumeSnapshotContent
volumesnapshots                   vs                  snapshot.storage.k8s.io/v1             true         VolumeSnapshot

```

Kubernetes version:
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.5", GitCommit:"5c99e2ac2ff9a3c549d9ca665e7bc05a3e18f07e", GitTreeState:"clean", BuildDate:"2021-12-16T08:38:33Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:19:12Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"linux/amd64"}
```
Velero version: 1.8.1
velero-plugin-for-csi version: 0.2.0
Rook version: 1.8.3

**Comment by sseago** [comment, marked as answer]
You probably only have snapshot.storage.k8s.io/v1 installed, not v1beta1. I just hit the same issue with 1.8.1. The 1.8.z releases look for v1beta1. The upcoming 1.9 release looks for v1. I installed velero from current main branch and this problem went away (but that's using an unreleased/unsupported velero build). You can probably fix this on the other end by figuring out how to install the v1beta1 version of the CSI infrastructure instead of v1 and still use velero 1.8.1

**Comment by avekrivoy** [comment]
I think you're right. I pulled Velero docker image tagged main and the error is gone. Still testing it, though.
1.9 is referenced in the Velero repository docs like it's been already released. Confused me a bit. Well, will wait for the official release

**Comment by sseago** [comment]
We expect to release 1.9 later this month.
